### PR TITLE
feat: added the strictNullChecks rule to tsconfig

### PR DESCRIPTION
**Pull request by Aleksey28**
## Purpose
Synchronize typescript options with testcafe for monorepo migration.

## Approach
Added the strictNullChecks rule to tsconfig, fixed errors.

## References
Part of https://github.com/DevExpress/testcafe/issues/5666

## Pre-Merge TODO
- [x] Make sure that existing tests do not fail

### CODE REVIEW

File: src/processing/script/transformers/assignment-destructuring.ts [hunk 1]
```diff
@@ -26,20 +26,22 @@ const transformer: Transformer<AssignmentExpression> = {
     condition: node => node.operator === '=' && (node.left.type === Syntax.ObjectPattern || node.left.type === Syntax.ArrayPattern),
 
     run: (node, _parent, _key, tempVars) => {
-        const assignments = [] as (AssignmentExpression | Identifier)[];
-        let firstTemp     = void 0;
+        const assignments = [] as (MemberExpression | AssignmentExpression | Pattern)[];
```
Comment (LavrovArtem) [inline src/processing/script/transformers/assignment-destructuring.ts:29]: Why do we need this change?

File: src/processing/script/transformers/temp-variables.ts [hunk 2]
```diff
@@ -11,15 +11,15 @@ export default class TempVariables {
         TempVariables._counter = 0;
     }
 
-    static generateName(baseName?: string, key?: Expression, index?: number): string {
+    static generateName(baseName?: string, key?: Expression | null, index?: number): string {
```
Comment (LavrovArtem) [inline src/processing/script/transformers/temp-variables.ts:14]: Unnecessary

File: src/proxy/router.ts [hunk 3]
```diff
@@ -57,8 +57,8 @@ export default abstract class Router {
     }
 
     _prepareParamInfo (tokens: string[], method: string) {
-        const paramNames = [];
-        const reParts    = tokens.map(token => {
+        const paramNames: string[] = [];
+        const reParts              = tokens.map(token => {
```
Comment (LavrovArtem) [inline src/proxy/router.ts:61]:
```suggestion
        const paramNames = [] as string[];
        const reParts    = tokens.map(token => {
```

File: src/request-pipeline/context.ts [hunk 4]
```diff
@@ -29,6 +28,7 @@ import requestIsMatchRule from '../request-pipeline/request-hooks/request-is-mat
 import getMockResponse from '../request-pipeline/request-hooks/response-mock/get-response';
 import { Http2Response } from './destination-request/http2';
 import RequestEvent from '../session/events/request-event';
+import net from 'net';
```
Comment (LavrovArtem) [inline src/request-pipeline/context.ts:31]: Unnecessary change

**Review by LavrovArtem — CHANGES_REQUESTED** [review]

File: src/processing/script/transformers/assignment-destructuring.ts (hunk 1 above)
Comment (Aleksey28) [reply src/processing/script/transformers/assignment-destructuring.ts:29]: I tried to link the 'firstTemp' changes to 'assignments' according to TS.

**Review by LavrovArtem — CHANGES_REQUESTED** [review]

File: src/processing/encoding/index.ts [hunk 5]
```diff
@@ -41,11 +41,11 @@ export async function decodeContent (content: Buffer, encoding: string, charset:
 
     charset.fromBOM(content);
 
-    return charsetEncoder.decode(content, charset.get());
+    return charsetEncoder.decode(content, charset.get() || '');
```
[the remark this answers is not in the document]
Comment (Aleksey28) [reply src/processing/encoding/index.ts:44]: Because the encoding has been undefined before. Now it unnecessary.

**Review by LavrovArtem — APPROVED** [review]

**Review by LavrovArtem — APPROVED** [review]

**Review by AndreyBelym — APPROVED** [review]
